docs/manual-EN/3.build-develop-and-administration/1.build/1.build-source-code.md

```diff
@@ -1,244 +1,200 @@
-bash> cp etc/nebula-metad.conf.default etc/nebula-metad.conf
-bash> cp etc/nebula-storaged.conf.default etc/nebula-storaged.conf
-```
+#### ENABLE_CCACHE
+This option is to enable the use of `ccache`, which is for speeding up the compilation during daily development.
 
-Start Nebula Graph services.
+By default, **Nebula Graph** will take advantage of `ccache` if it's found. So you don't have to enable it for yourself.
 
-```bash
-bash> ./scripts/nebula.service start all
-bash> ./bin/nebula -u user -p password --port 3699 --addr="127.0.0.1"
-nebula> SHOW HOSTS;
-```
+If you want to disable `ccache`, it might be not enough to just turn `ENABLE_CCACHE` off. Since on some platforms, the `ccache` installation hooks up or precedes the compiler. For such an case, you have to set an evnironment variable `export CCACHE_DISABLE=true`, or add a line `disable=true` to `~/.ccache/ccache.conf`. We may do this for you automatically in future.
 
-## Trouble Shooting
+Please see the [official documentation](https://ccache.dev/manual/3.7.6.html) for more details on it.
 
-### Error Info: `/usr/bin/ld: cannot find Scrt1.o: No such file or directory`
+#### NEBULA_USE_LINKER
+This option is for allowing users to use an alternative linker, e.g. `gold`. Optionals are `bfd` and `gold` for now.
 
-1. Modify **~/.bashrc** by appending following line to the end
+#### NEBULA_THIRDPARTY_ROOT
+This option is to explicitly specify the location of the third party.
 
+### To Install Third Party Manually
+By default, at the configure(cmake) stage, a prebuilt third party will be downloaded and installed to the current build directory. If you would like to install it into another location for some reason, e.g. to rebuild by removing the whole build directory without download the third party again, you could perform the install manually. Assuming you are now at the build diretory, run:
 ```bash
-export LIBRARY_PATH=/usr/lib/x86_64-linux-gnu:$LIBRARY_PATH
+# To install third party to /opt requires root privilege, you could change it to another location with --prefix.
+$ ../third-party/install-third-party.sh --prefix=/opt/vesoft/third-party
 ```
 
-2. Update **~/.bashrc**
+If the third party is installed to `/opt/vesoft/third-party`, which is by default if no `--prefix` given, the building system of **Nebula Graph** would find it automatically. Otherwise, you need specify the location with the CMake argument `NEBULA_THIRDPARTY_ROOT` as mentioned above, or set an environment variable to the location and export it. The precedence for **Nebula Graph** to hunt for the third party is:
+ 1. The CMake argument `NEBULA_THIRDPARTY_ROOT`
+ 2. `third-party/install` in the current build directory
+ 3. The `NEBULA_THIRDPARTY_ROOT` environment variable
+ 4. `/opt/vesoft/third-party`
 
+## Install an Applicable CMake
+For users who don't have an usable CMake installation, we provide a script to automatically download and install one for you. Assuming you are now at the build directory, run:
```

Comment:
have an -> have a

Reply:
I think _an_ is correct.

Reply:
usable /juː.zə.bəl/, so wilson is right.

Reply:
you are right.